environment/mel_env.py

```diff
@@ -49,3 +49,17 @@ def step(self, action, sr=41_000):
         magnitude = np.array(s_full)
         next_state = np.sum(magnitude, axis=1)/magnitude.shape[1]
         return next_state, reward, terminated, truncated, info
+
+    def perform_attack(self, action, audio, sr=41_000):
```

Comment:
Really like this, would be nice if we used it in the step function, though I would be okay if we did this another time.